File: backend/internal/service/system/admin/v1/ops_monitoring_sse.go
package admin

import coreSSE "github.com/liujitcn/kratos-admin/backend/core/pkg/sse"

const (
	// OpsMonitoringSSEStreamID 表示运维监控实时数据流。
	OpsMonitoringSSEStreamID = "system.admin.ops-monitoring"
	// OpsMonitoringSSETraffic 表示流量监控事件。
	OpsMonitoringSSETraffic = "ops.traffic"
	// OpsMonitoringSSEServices 表示服务状态事件。
	OpsMonitoringSSEServices = "ops.services"
	// OpsMonitoringSSEStorage 表示存储状态事件。
	OpsMonitoringSSEStorage = "ops.storage"
	// OpsMonitoringSSENodes 表示实例资源事件。
	OpsMonitoringSSENodes = "ops.nodes"
)

// OpsMonitoringSSEStream 声明运维监控固定 SSE 流。
type OpsMonitoringSSEStream struct{}

var _ coreSSE.Stream = (*OpsMonitoringSSEStream)(nil)

// NewOpsMonitoringSSEStream 创建运维监控 SSE 流声明。
func NewOpsMonitoringSSEStream() *OpsMonitoringSSEStream {
	return &OpsMonitoringSSEStream{}
}

// ID 返回运维监控 SSE 流标识。
func (*OpsMonitoringSSEStream) ID() string {
	return OpsMonitoringSSEStreamID
}

// Resolve 返回固定的运维监控传输流标识。
func (*OpsMonitoringSSEStream) Resolve(string, int64) (string, error) {
	return OpsMonitoringSSEStreamID, nil
}

File: frontend/admin/packages/modules/system/src/utils/ops_monitoring_sse.ts
import { subscribeSseEvent, type SseStop } from "../api/base/v1/sse";
import type {
  OpsNodesResponse,
  OpsServicesResponse,
  OpsStorageResponse,
  OpsTrafficResponse
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/ops_monitoring";

const OPS_MONITORING_SSE_STREAM = "system.admin.ops-monitoring";
const OPS_MONITORING_SSE_TRAFFIC = "ops.traffic";
const OPS_MONITORING_SSE_SERVICES = "ops.services";
const OPS_MONITORING_SSE_STORAGE = "ops.storage";
const OPS_MONITORING_SSE_NODES = "ops.nodes";

/** 运维监控 SSE 取消订阅函数。 */
export type OpsMonitoringSseStop = SseStop;

/** 订阅运维监控流量实时事件。 */
export function subscribeOpsMonitoringTraffic(handler: (payload: OpsTrafficResponse) => void): SseStop {
  return subscribeOpsMonitoringEvent(OPS_MONITORING_SSE_TRAFFIC, parsePayload<OpsTrafficResponse>, handler);
}

/** 订阅运维监控服务实时事件。 */
export function subscribeOpsMonitoringServices(handler: (payload: OpsServicesResponse) => void): SseStop {
  return subscribeOpsMonitoringEvent(OPS_MONITORING_SSE_SERVICES, parsePayload<OpsServicesResponse>, handler);
}

/** 订阅运维监控存储实时事件。 */
export function subscribeOpsMonitoringStorage(handler: (payload: OpsStorageResponse) => void): SseStop {
  return subscribeOpsMonitoringEvent(OPS_MONITORING_SSE_STORAGE, parsePayload<OpsStorageResponse>, handler);
}

/** 订阅运维监控实例实时事件。 */
export function subscribeOpsMonitoringNodes(handler: (payload: OpsNodesResponse) => void): SseStop {
  return subscribeOpsMonitoringEvent(OPS_MONITORING_SSE_NODES, parsePayload<OpsNodesResponse>, handler);
}

function subscribeOpsMonitoringEvent<T>(event: string, parser: (raw: string) => T | null, handler: (payload: T) => void): SseStop {
  return subscribeSseEvent({ stream: OPS_MONITORING_SSE_STREAM }, event, parser, handler);
}

function parsePayload<T>(raw: string): T | null {
  if (!raw) return null;
  try {
    return JSON.parse(raw) as T;
  } catch {
    return null;
  }
}
